Scripts/PAIF-N/01-deploy-vcf-workload-domain.ps1

```diff
@@ -0,0 +1,345 @@
+               Gateway = $gateway
+            }
+         }
+         @{
+            Name = "$domainName-nsx01c"
+            NetworkDetailsSpec = @{
+               DnsName = "$domainName-nsx01c.$domain"
+               IpAddress = "10.0.0.43"
+               SubnetMask = "255.255.255.0"
+               Gateway = $gateway
+            }
+         }
+      )
+      FormFactor = 'large'
+      LicenseKey = 'XXXXX-XXXXX-XXXXX-XXXXX-XXXXX'
+      NSXManagerAdminPassword = "VMware123!123"
+      Vip = '10.0.0.44'
+      VipFqdn = "$domainName-nsx01.$domain"
+   }
+}
+# --------------------------------------------------------------------------------------------------------------------------
+
+## Workload domain spec construction
+$esxiFqdns = $esxiHosts | Select-Object -ExpandProperty Fqdn
+$VcfHost = Invoke-VcfGetHosts -status "UNASSIGNED_USEABLE" | `
+   Select-Object -ExpandProperty Elements | `
+   Where-Object { $esxiFqdns -contains $_.Fqdn }
+
+$ComputeSpec = Initialize-VcfComputeSpec -ClusterSpecs (
+   $domainSpec.ComputeSpec.ClusterSpecs | ForEach-Object {
+      $_clusterSpec = $_
+
+      Initialize-VcfClusterSpec `
+         -DatastoreSpec (
+            Initialize-VcfDatastoreSpec `
+               -VsanDatastoreSpec (
+                  Initialize-VcfVsanDatastoreSpec `
+                     -DatastoreName $_.DatastoreSpec.VSanDatastoreSpec.DatastoreName `
+                     -FailuresToTolerate $_.DatastoreSpec.VSanDatastoreSpec.FailuresToTolerate `
+                     -LicenseKey $_.DatastoreSpec.VSanDatastoreSpec.LicenseKey
+               )) `
+         -HostSpecs (
+            $VcfHost | ForEach-Object {
+               $esxiHost = $esxiHosts | Where-Object { $_.Fqdn -eq $_.Fqdn } | Select-Object -First 1
+
+               Initialize-VcfHostSpec `
+                  -Id $_.Id `
+                  -LicenseKey $esxiHost.LicenseKey `
+                  -HostNetworkSpec (
+                     Initialize-VcfHostNetworkSpec -VmNics @(
+                        Initialize-VcfVmNic -Id 'vmnic0' -VdsName $_clusterSpec.NetworkSpec.VdsSpecs[0].Name
+                        Initialize-VcfVmNic -Id 'vmnic1' -VdsName $_clusterSpec.NetworkSpec.VdsSpecs[0].Name
+                        Initialize-VcfVmNic -Id 'vmnic2' -VdsName $_clusterSpec.NetworkSpec.VdsSpecs[1].Name
+                        Initialize-VcfVmNic -Id 'vmnic3' -VdsName $_clusterSpec.NetworkSpec.VdsSpecs[1].Name
+                     ))
+            }
+         ) `
+         -Name $_.Name `
+         -NetworkSpec (
+            Initialize-VcfNetworkSpec `
+               -NsxClusterSpec (
+                  Initialize-VcfNsxClusterSpec -NsxTClusterSpec (
+                     Initialize-VcfNsxTClusterSpec `
+                        -GeneveVlanId $_.NetworkSpec.NsxClusterSpec.NsxTClusterSpec.GeneveVlanId
+                        )
+                  ) `
+               -VdsSpecs @(
+                  Initialize-VcfVdsSpec -Name $_.NetworkSpec.VdsSpecs[0].Name `
```

Comment:
Same as above, does this create 2 distributed switches, can this be relaxed?

Reply:
I've documented in the prerequisites.